encoder_borsh.go

```diff
@@ -70,35 +119,15 @@ func (e *Encoder) encodeBorsh(rv reflect.Value, opt *option) (err error) {
 		return marshaler.MarshalWithEncoder(e)
 	}
 
+	// Encode the value if it's a primitive type
+	isPrimitive, err := e.encodePrimitive(rv, nil)
+	if isPrimitive {
+		return
+	} else if err != nil {
+		return err
+	}
+
```

Comment:
`err` is returned only if `isPrimitive == true`

You also shadowed `err` with `:=`

```suggestion
	// Encode the value if it's a primitive type
	isPrimitive, err := e.encodePrimitive(rv, nil)
	if isPrimitive {
		return err
	} 

```

Reply:
Also not sure what you mean here: "You also shadowed err with :=" 

Could you elaborate + suggest a fix? Thanks!

Reply:
If the `encodePrimitive` function returns `isPrimitive == false`, it means that no write operation was executed and it cannot return any errors.

Reply:
If `isPrimitive == true`, this means a write operation was executed and only in that case `err` should be checked.

Reply:
Also, if `isPrimitive == true`, the parent must exit returning `err` just like it did when the primitive switch cases were in the bigger switch case below.

Reply:
> Also not sure what you mean here: "You also shadowed err with :="

Ignore that.

Reply:
I've changed the code as requested :+1: